lxd/db/oidc/session.go

```diff
@@ -0,0 +1,240 @@
+package oidc
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/netip"
+	"time"
+
+	"github.com/google/uuid"
+	zitadelOIDC "github.com/zitadel/oidc/v3/pkg/oidc"
+	"golang.org/x/oauth2"
+
+	"github.com/canonical/lxd/lxd/auth/oidc"
+	"github.com/canonical/lxd/lxd/db"
+	"github.com/canonical/lxd/lxd/db/cluster"
+	"github.com/canonical/lxd/lxd/events"
+	"github.com/canonical/lxd/lxd/lifecycle"
+	"github.com/canonical/lxd/lxd/request"
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/lxd/shared/api"
+	"github.com/canonical/lxd/shared/logger"
+)
+
+// NewSessionHandler returns a new [oidc.SessionHandler]. The getSessionExpiry function must return the current value
+// of oidc.session.expiry for the server configuration.
+func NewSessionHandler(db *db.Cluster, events *events.Server, getSessionExpiry func() string) oidc.SessionHandler {
```

Comment:
seems like we can just return `sessionHandler` here and only make the fields/functions exported that are in the interface.